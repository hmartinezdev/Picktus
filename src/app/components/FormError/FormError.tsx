import ErrorSvg from '@assets/svg/error.svg';
import colors from '@constants/colors';
import React, { Component } from 'react';

export interface IFormErrorPropsType {
  text: string;
}
class FormError extends Component<IFormErrorPropsType> {
  public render(): React.ReactElement<FormError> {
    const { text } = this.props;

    return (
      <div className="container">
        <ErrorSvg className="icon" />
        <p>{text}</p>
        <style jsx>{`
          p {
            line-height: 1rem;
            margin-top: 0.2rem;
            box-sizing: border-box;
          }

          .container {
            display: flex;
            align-items: center;
            font-family: 'Josefin Sans', sans-serif;
            font-size: 0.7rem;
            color: ${colors.white};
            width: 100%;
            box-sizing: border-box;
            letter-spacing: 0.05rem;
            padding: 0.5rem 0rem 0.5rem 0.2rem;
            transition: 0.4s;
            outline: none;
            background-color: ${colors.error}cd;
            border-radius: 3px;
            margin: 0.8rem 0;
          }

          .icon {
            fill: ${colors.secondary};
            min-width: 1.8rem;
            margin-right: 0.2rem;
          }
        `}</style>
      </div>
    );
  }
}

export default FormError;
