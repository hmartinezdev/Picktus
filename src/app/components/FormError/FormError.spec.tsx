import { shallow } from 'enzyme';
import React from 'react';
import FormError from './FormError';

let wrapper;
const setup = (props = { text: 'erreur' }) => shallow(<FormError {...props} />);

describe('<FormError />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
