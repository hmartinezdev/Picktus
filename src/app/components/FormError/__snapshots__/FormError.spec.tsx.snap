// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormError /> should render properly 1`] = `
<div
  className="container"
>
  <Component
    className="icon"
  />
  <p>
    erreur
  </p>
  <style
    jsx={true}
  >
    
          p {
            line-height: 1rem;
            margin-top: 0.2rem;
            box-sizing: border-box;
          }

          .container {
            display: flex;
            align-items: center;
            font-family: 'Josefin Sans', sans-serif;
            font-size: 0.7rem;
            color: #F8F8F7;
            width: 100%;
            box-sizing: border-box;
            letter-spacing: 0.05rem;
            padding: 0.5rem 0rem 0.5rem 0.2rem;
            transition: 0.4s;
            outline: none;
            background-color: #982649cd;
            border-radius: 3px;
            margin: 0.5rem 0;
          }

          .icon {
            fill: #FAC9B8;
            min-width: 1.8rem;
            margin-right: 0.2rem;
          }
        
  </style>
</div>
`;
